ci-operator/step-registry/ipi/conf/aws/user-min-permissions/ipi-conf-aws-user-min-permissions-commands.sh

```diff
@@ -7,33 +7,57 @@ set -o pipefail
 trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
 trap 'rm -rf /tmp/aws_cred_output /tmp/pull-secret /tmp/min_perms/ /tmp/jsoner.py' EXIT TERM INT
 
-USER_POLICY_FILENAME="aws-permissions-policy-creds.json"
-USER_POLICY_FILE="${SHARED_DIR}/${USER_POLICY_FILENAME}"
 
-if [[ "${AWS_INSTALL_USE_MINIMAL_PERMISSIONS}" != "yes" ]]; then
-	echo "Custom AWS user with minimal permissions is disabled. Using AWS user from cluster profile."
-	exit 0
-fi
+JSONER_PY="/tmp/jsoner.py"
 
-RELEASE_IMAGE_INSTALL="${RELEASE_IMAGE_INITIAL:-}"
-if [[ -z "${RELEASE_IMAGE_INSTALL}" ]]; then
-	# If there is no initial release, we will be installing latest.
-	RELEASE_IMAGE_INSTALL="${RELEASE_IMAGE_LATEST:-}"
-fi
-cp ${CLUSTER_PROFILE_DIR}/pull-secret /tmp/pull-secret
-oc registry login --to /tmp/pull-secret
-ocp_version=$(oc adm release info --registry-config /tmp/pull-secret ${RELEASE_IMAGE_INSTALL} -ojsonpath='{.metadata.version}' | cut -d. -f 1,2)
-ocp_major_version=$(echo "${ocp_version}" | awk --field-separator=. '{print $1}')
-ocp_minor_version=$(echo "${ocp_version}" | awk --field-separator=. '{print $2}')
-rm /tmp/pull-secret
+function create_jsoner_py()
+{
+	if [[ ! -f ${JSONER_PY} ]]; then
+		cat <<EOF >"${JSONER_PY}"
+import json
+import sys
+
+data = sys.stdin.read().splitlines()
+out = {
+	"Version": "2012-10-17",
+	"Statement": [{
+		"Effect": "Allow",
+		"Resource": "*",
+		"Action": data,
+	}]
+}
+print(json.dumps(out, indent=2))
+EOF
+	fi
+}
 
-export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+if [[ "${AWS_INSTALL_USE_MINIMAL_PERMISSIONS}" == "yes" ]]; then
 
-if ((ocp_major_version < 4 || (ocp_major_version == 4 && ocp_minor_version < 18))); then
-	# There is no installer support for generating permissions prior to 4.18, so we generate one ourselves
-	PERMISION_LIST="${ARTIFACT_DIR}/permision_list.txt"
+	export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
```

Comment:
Seem like we do not need this ENV any more.

Reply:
It's required by `openshift-install create permissions-policy --dir ${dir}`

Reply:
I tried locally, seem like no.

Reply:
They are required by some particular configuration:
like [this](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/58859/rehearse-58859-pull-ci-openshift-installer-release-4.18-e2e-aws-ovn-shared-vpc-custom-security-groups/1859604184899260416)
```
 INFO[2024-11-21T16:02:03Z] info: Using registry public hostname registry.build09.ci.openshift.org
Saved credentials for registry.build09.ci.openshift.org into /tmp/pull-secret
? AWS Access Key ID [? for help] [?25l[?25l78[?25h[?25hlevel=error msg=failed to fetch Permissions: failed to load asset "Install Config": failed to create install config: unable to load edge subnets: error retrieving Edge Subnets: creating AWS session: EOF: error retrieving Edge Subnets: creating AWS session: EOF
{"component":"entrypoint","error":"wrapped process failed: exit status 3","file":"sigs.k8s.io/prow/pkg/entrypoint/run.go:84","func":"sigs.k8s.io/prow/pkg/entrypoint.Options.internalRun","level":"error","msg":"Error executing test process","severity":"error","time":"2024-11-21T16:02:02Z"}
error: failed to execute wrapped command: exit status 3 
```

and [this](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/58859/rehearse-58859-pull-ci-openshift-installer-release-4.18-e2e-aws-ovn/1859604192440619008):
```
 INFO[2024-11-21T15:28:47Z] info: Using registry public hostname registry.build09.ci.openshift.org
Saved credentials for registry.build09.ci.openshift.org into /tmp/pull-secret
level=warning msg=Release Image Architecture not detected. Release Image Architecture is unknown
? AWS Access Key ID [? for help] [?25l[?25l78[?25h[?25h? AWS Access Key ID [? for help] [?25l[?25l78[?25h[?25h? AWS Access Key ID [? for help] [?25l[?25l78[?25h[?25hlevel=error msg=failed to fetch Permissions: failed to load asset "Install Config": failed to create install config: [platform.aws.publicIpv4PoolId: Internal error: creating AWS session: EOF, controlPlane.platform.aws: Internal error: creating AWS session: EOF, compute[0].platform.aws: Internal error: creating AWS session: EOF]
{"component":"entrypoint","error":"wrapped process failed: exit status 3","file":"sigs.k8s.io/prow/pkg/entrypoint/run.go:84","func":"sigs.k8s.io/prow/pkg/entrypoint.Options.internalRun","level":"error","msg":"Error executing test process","severity":"error","time":"2024-11-21T15:28:46Z"}
error: failed to execute wrapped command: exit status 3 
```

Reply:
Good to know, thanks.

Reply:
Yeah, that's why this command is hidden and supposed to be used internally-only. Validating the `install-config.yaml` might involve aws sdk calls which require certain permissions.